akka-persistence/src/main/scala/akka/persistence/state/scaladsl/DurableStateStore.scala

```diff
@@ -19,4 +20,6 @@ trait DurableStateStore[A] {
 
 }
 
-final case class GetObjectResult[A](value: Option[A], revision: Long)
+final case class GetObjectResult[A](value: Option[A], revision: Long) {
+  def toJava = akka.persistence.state.javadsl.GetObjectResult(value.asJava, revision)
```

Comment:
Always use explicit return type for public api. Reason is to make sure that the public api has the right types and not something accidental from the implementation. Also prevents from accidental binary compatibility breakage in the future. Mima might detect that but it's an additional safety.